pylint/extensions/check_mccabe.py

```diff
@@ -0,0 +1,176 @@
+        self.tail = node
+        return node
+
+    def _subgraph(self, node, name, extra_blocks=()):
+        """create the subgraphs representing any `if` and `for` statements"""
+        if self.graph is None:
+            # global loop
+            self.graph = PathGraph(node)
+            self._subgraph_parse(node, node, extra_blocks)
+            self.graphs["%s%s" % (self.classname, name)] = self.graph
+            self.reset()
+        else:
+            self._append_node(node)
+            self._subgraph_parse(node, node, extra_blocks)
+
+    def _subgraph_parse(self, node, pathnode, extra_blocks):  # pylint: disable=unused-argument
+        """parse the body and any `else` block of `if` and `for` statements"""
+        loose_ends = []
+        self.tail = node
+        self.dispatch_list(node.body)
+        loose_ends.append(self.tail)
+        for extra in extra_blocks:
+            self.tail = node
+            self.dispatch_list(extra.body)
+            loose_ends.append(self.tail)
+        if node.orelse:
+            self.tail = node
+            self.dispatch_list(node.orelse)
+            loose_ends.append(self.tail)
+        else:
+            loose_ends.append(node)
+        if node:
+            bottom = "%s" % self._bottom_counter
+            self._bottom_counter += 1
+            for le in loose_ends:
+                self.graph.connect(le, bottom)
+            self.tail = bottom
+
+
+class McCabeMethodChecker(BaseChecker):
+    """Checks McCabe complexity cyclomatic threshold in methods and functions
+    to validate a too complex code.
+    """
+
+    __implements__ = IAstroidChecker
+    name = 'design'
+
+    msgs = {
+        'R1260': (
+            '%s is too complex. The McCabe rating is %d',
+            'too-complex',
+            'Used when a method or function is too complex based on '
+            'McCabe Complexity Cyclomatic'),
+    }
+    options = (
+        ('max-complexity', {
+            'default': 10,
+            'type': 'int',
+            'metavar': '<int>',
+            'help': 'McCabe complexity cyclomatic threshold',
+        }),
+    )
+
+    def _check_too_complex(self, node):
+        """Check too complex rating and
+        add message if is greather than max_complexity stored from options"""
+        visitor = PathGraphingAstVisitor()
+        for child in node.body:
+            visitor.preorder(child, visitor)
+        for graph in visitor.graphs.values():
+            complexity = graph.complexity()
+            node = graph.root
+            if node.is_function:
+                node_name = node.name
+            else:
+                node_name = "'%s %d'" % (node.__class__.__name__, node.lineno)
+            if complexity <= self.config.max_complexity:
+                continue
+            self.add_message(
+                'too-complex', node=node, confidence=HIGH,
+                args=(node_name, complexity))
+
+    @check_messages('too-complex')
+    def visit_module(self, node):
```

Comment:
Isn't this the same as _check_too_complex? Do you plan to use _check_too_complex somewhere else? Otherwise we could just rename _check_too_complex to visit_module.